astropy/utils/misc.py

```diff
@@ -382,18 +384,43 @@ class JsonCustomEncoder(json.JSONEncoder):
 
     """
     def default(self, obj):
+        from .. import units as u
         import numpy as np
-        if isinstance(obj, (np.ndarray, np.number)):
+        if isinstance(obj, np.number):
+            return obj.tolist()
+        elif isinstance(obj, u.Quantity):
+            return dict(value=obj.value, unit=obj.unit.to_string())
```

Comment:
Is this `dict` called recursively? If so, there would seem to be no need to use `to_string` on the unit -- that would automatically get done in the recursion.